core/templates/services/math-interactions.service.ts

```diff
@@ -374,6 +374,54 @@ export class MathInteractionsService {
 
     return partsList1.length === 0 && partsList2.length === 0;
   }
+
+  expressionMatchWithPlaceholders(
+      expressionWithPlaceholders: string, expressionWithoutPlaceholders: string,
```

Comment:
Maybe rename the second var to inputExpression or something that specifies its use? Since the current name doesn't indicate why we need placeholders.

Reply:
Agreed, done.